lightning/restore/restore.go

```diff
@@ -1696,23 +1744,35 @@ type RemoteChecksum struct {
 	TotalBytes uint64
 }
 
-// DoChecksum do checksum for tables.
-// table should be in <db>.<table>, format.  e.g. foo.bar
-func DoChecksum(ctx context.Context, db *sql.DB, table string) (*RemoteChecksum, error) {
-	var err error
-	manager, ok := ctx.Value(&gcLifeTimeKey).(*gcLifeTimeManager)
-	if !ok {
-		return nil, errors.New("No gcLifeTimeManager found in context, check context initialization")
+type ChecksumManager interface {
```

Comment:
perhaps all two entire checksum managers into another file. this file is too large 😂